app/utils/s3.py

```diff
@@ -0,0 +1,128 @@
+"""
+S3 client handler.
+"""
+from pathlib import Path
+import boto3
+import botocore
+import utils.settings as settings
+
+
+class S3Handler:  # pylint: disable=too-many-instance-attributes
+    """
+    Encapsulates the R/W logic from/to a S3 endpoint.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        bucket,
+        endpoint,
+        region,
+        secret_key,
+        access_key,
+        verify,
+        cert,
+        private_key,
+    ):
+        """
+        Constructor for initialising a S3 client.
+        """
+        assert isinstance(bucket, str)
+        self.bucket = bucket
+        assert isinstance(endpoint, str)
+        self.endpoint = endpoint
+        assert isinstance(region, str)
+        self.region = region
+        assert isinstance(access_key, str)
+        self.access_key = access_key
+        assert isinstance(secret_key, str)
+        self.secret_key = secret_key
+        assert isinstance(verify, (bool, Path))
+        self.verify = verify
+        assert isinstance(cert, (Path, type(None)))
+        self.cert = cert
+        assert isinstance(private_key, (Path, type(None)))
+        self.private_key = private_key
+
+        config_params = {
+            "connect_timeout": 40,
+        }
+
+        if cert and private_key:
+            config_params["client_cert"] = (cert, private_key)
+
+        config = botocore.client.Config(**config_params)
+        self.s3_client = boto3.client(
+            "s3",
+            endpoint_url=self.endpoint,
+            region_name=region,
+            use_ssl=self.endpoint.startswith("https"),
+            verify=verify,
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            config=config,
+        )
+
+        print(vars(self))
```

Comment:
These are easy to forget, better to write a suitable log statement directly?

Reply:
Yes. Good that you found this one. We can set it as a trace or debug line.